draft-ietf-tls-esni.md

```diff
@@ -686,72 +684,72 @@ Offering a GREASE extension is not considered offering an encrypted ClientHello
 for purposes of requirements in {{client-behavior}}. In particular, the client
 MAY offer to resume sessions established without ECH.
 
-# Client-Facing Server Behavior {#server-behavior}
+# Server Behavior {#server-behavior}
+
+## Client-Facing Server
 
 Upon receiving an "encrypted_client_hello" extension, the client-facing server
 MUST check that it is able to negotiate TLS 1.3 or greater. If not, it MUST
 abort the connection with a "handshake_failure" alert.
 
-The ClientEncryptedCH value is said to match a known ECHConfig if there exists
+The ClientECH value is said to match a known ECHConfig if there exists
 an ECHConfig that can be used to successfully decrypt
-ClientEncryptedCH.encrypted_ch. This matching procedure should be done using
+ClientECH.encrypted_ch. This matching procedure should be done using
 one of the following two checks:
 
-1. Compare ClientEncryptedCH.config_id against identifiers of known ECHConfig
+1. Compare ClientECH.config_id against identifiers of known ECHConfig
    and choose the one that matches.
-2. Use trial decryption of ClientEncryptedCH.encrypted_ch with known ECHConfig
+2. Use trial decryption of ClientECH.encrypted_ch with known ECHConfig
    and choose the one that succeeds.
 
-Some uses of ECH, such as local discovery mode, may omit the
-ClientEncryptedCH.config_id since it can be used as a tracking vector. In
-such cases, trial decryption should be used for matching ClientEncryptedCH to
-known ECHConfig. Unless specified by the application using (D)TLS or externally
-configured on both sides, implementations MUST use the first method.
+Some uses of ECH, such as local discovery mode, may omit the ClientECH.config_id
+since it can be used as a tracking vector. In such cases, trial decryption
+should be used for matching ClientECH to known ECHConfig. Unless specified by
+the application using (D)TLS or externally configured on both sides,
+implementations MUST use the first method.
 
-If the ClientEncryptedCH value does not match any known ECHConfig structure, it
+If the ClientECH value does not match any known ECHConfig structure, it
 MUST ignore the extension and proceed with the connection, with the following
 added behavior:
 
-- It MUST include the "encrypted_client_hello" extension with the
-  "retry_configs" field set to one or more ECHConfig structures with up-to-date
-  keys. Servers MAY supply multiple ECHConfig values of different versions. This
-  allows a server to support multiple versions at once.
+- It SHOULD include the "encrypted_client_hello" extension in its
```

Comment:
Please revert this change -- this MUST was there so clients could rely on this particular version of ECH being deployable. (We can discuss it in a separate issue.)

```suggestion
- It MUST include the "encrypted_client_hello" extension in its
```

Reply:
Ack. Fixing in #287, then rebasing here.